### PR TITLE
fix sharded tests

### PR DESCRIPTION
**Pull request by daprahamian**
Fixes and/or removes any issues with our sharding tests, so that we can actually pass sharding tests in evergreen.

### CODE REVIEW

File: test/functional/find_tests.js [hunk 1]
```diff
@@ -2629,7 +2629,7 @@ describe('Find', function() {
     metadata: {
       requires: {
         mongodb: '>=3.6.0',
-        topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger']
+        topology: ['single', 'replicaset']
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply test/functional/find_tests.js]: From the docs: https://docs.mongodb.com/manual/reference/command/parallelCollectionScan/

> parallelCollectionScan is only available for mongod, and it cannot operate on a sharded cluster.

I also got rid of SSL, Heap, and Wiredtiger b/c we don't really use them.

File: test/functional/sessions_tests.js [hunk 2]
```diff
@@ -10,19 +9,37 @@ describe('Sessions', function() {
     return setupDatabase(this.configuration);
   });
 
-  afterEach(() => test.listener.uninstrument());
+  function parseAPMEvent(e) {
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply test/functional/sessions_tests.js]: I think that is a good question. On the one hand, I'd like APM events to be uniform. On the other, I believe this is a side effect of OP_QUERY in specific circumstances which have led to bugs in the past, and those bugs were surfaced by showing that the APM command was different.

File: test/functional/sharding_connection_tests.js [hunk 3]
```diff
@@ -67,23 +55,20 @@ describe('Sharding (Connection)', function() {
...
-        setTimeout(function() {
-          test.equal(null, err);
-          client.close();
-          done();
-        }, 2000);
+      const configuration = this.configuration;
+      const host = configuration.host;
+      const port = configuration.port;
+
+      // TODO: Better way to do this?
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply test/functional/sharding_connection_tests.js]: Probably. I was trying to get it to work with as little work as possible. If your refactor to use connection strings already did this, that should be fine as well.

File: test/functional/sharding_read_preference_tests.js [hunk 4]
```diff
@@ -15,29 +18,19 @@ describe('Sharding (Read Preference)', function() {
...
-        new Server(configuration.host, configuration.port + 1, { auto_reconnect: true })
-      ]);
+      const configuration = this.configuration;
 
       // Connect using the mongos connections
-      var client = new MongoClient(mongos, { w: 0 });
-      client.connect(function(err, client) {
-        test.equal(null, err);
-        var db = client.db(configuration.db);
+      var client = new MongoClient(configuration.url(), { w: 0 });
```
[the remark this answers is not in the document]
Comment (daprahamian) [reply test/functional/sharding_read_preference_tests.js]: [It appears that it does not](https://github.com/mongodb/node-mongodb-native/blob/master/test/environments.js#L134). We should probably fix that.

File: test/functional/sessions_tests.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mbroadst) [reply test/functional/sessions_tests.js]: It looks like I'm correct and by the APM implementation spec notes this should be converted to a `command`. See [this section](https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#implementation-notes)

**Comment by mbroadst** [comment]
Looks like we need conflicts resolved as well

**Pull request merged**